File: api/web/models/chat_log_session.py
"""APIs for ChatLogSession"""
from typing import Any, Dict, Optional
from modules.types import PMObject


class ChatLogInstance(PMObject):
    id: int
    created_at: str
    uuid: str
    run_from_deployment: bool

    session_metadata: Optional[Dict[str, Any]] = None

    version_uuid: str

File: src/types/ChatLogSession.ts
// Interfaces for ChatLogSession retrieval processes

/**
 * Interface for request to read ChatModelVersion's ChatLogSessions.
 */
export interface ReadChatLogSessionsRequest {
  chat_model_version_uuid: string;
}

/**
 * General interface for ChatLogSession.
 */
export interface ChatLogSession {
  id: number;
  created_at: string;
  uuid: string;
  run_from_deployment: boolean;
  session_metadata: Record<string, any>;
  version_uuid: string;
}
